src/main/java/seedu/address/logic/commands/AddPublicAddressCommand.java

```diff
@@ -0,0 +1,124 @@
+            + "wallet1 " + PREFIX_PUBLIC_ADDRESS + "1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa";
+
+    public static final String MESSAGE_ADDPA_SUCCESS = "Added Person's Public Address: %1$s";
+    public static final String MESSAGE_DUPLICATE_PUBLIC_ADDRESS =
+            "Invalid: person %1$s already has label %2$s under the network %3$s!\n"
+                    + "You may either:\n"
+                    + "1. Use another label for the new public address\n"
+                    + "2. Edit the existing public address for the current label"
+                    + "(overwrite) using the editpa command\n";
+
+    /**
+     * Takes in a target index and deletes all the addresses related to the network
+     *
+     * @param index
+     * @param editPersonDescriptor
+     */
+    public AddPublicAddressCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
+        super(index, editPersonDescriptor);
+    }
+
+    private static Person mergePersons(Person personToEdit, EditPersonDescriptor editPersonDescriptor) {
+        assert personToEdit != null;
+
+        Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
+        Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
+        Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
+        Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
+        Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
+
+        Map<Network, Set<PublicAddress>> currentPublicAddresses = new HashMap<>(personToEdit.getPublicAddresses());
+        Map<Network, Set<PublicAddress>> addedPublicAddresses =
+                editPersonDescriptor.getPublicAddresses().orElse(Map.of());
+
+        // even though this is a user input, checks should be done in the corresponding parser
+        assert addedPublicAddresses.size() == 1 : "Only one network should be added at a time";
+        assert addedPublicAddresses.values().toArray()[0] != null : "Public address should not be null";
+
+        // it is okay to suppress warnings here because editPersonDescriptor.getPublicAddresses() will always return
+        // a map with values of type Set<PublicAddress>
+        @SuppressWarnings("unchecked")
+        Set<PublicAddress> firstSet = (Set<PublicAddress>) addedPublicAddresses.values().toArray()[0];
+        assert firstSet.size() == 1 : "Only one public address should be added at a time";
+
+        Network network = (Network) addedPublicAddresses.keySet().toArray()[0];
+        PublicAddress publicAddress = firstSet.iterator().next();
+
+        // add network and public address to currentPubicAddresses
+        Set<PublicAddress> publicAddresses = new HashSet<>(currentPublicAddresses.getOrDefault(network, Set.of()));
+        if (publicAddresses.contains(publicAddress)) {
```

Comment:
Will this only return duplicate when the label and the network is the same? I thought we are only looking at label for unique fields?

Reply:
Agree with the future refactoring. Would be good to create a ticket for this, if haven't already.